sklearn/metrics/cluster/_supervised.py

```diff
@@ -1033,12 +1033,16 @@ def adjusted_mutual_info_score(
         or classes.shape[0] == clusters.shape[0] == 0
     ):
         return 1.0
+    # if there is only one class or one cluster return 0.0.
+    elif classes.shape[0] == 1 or clusters.shape[0] == 1:
+        return 0.0
```

Comment:
Is this special case tested in one of the already existing tests?

Reply:
There was a [failing](https://github.com/scikit-learn/scikit-learn/runs/39328593720) `test_exactly_zero_info_score` that tested exactly this case. Before this MR, AMI produced some around 0 scores (2e-16) in case of 1 cluster, that this test count as 0. After making clipping symmetric this test started to fail, since numerator value chaned. I added this shortcut to account for this edge case.

Maybe I can add more tests for this

Reply:
I am guessing the test could be updated to check that the value is exactly 0 rather than close to 0 in this case?